BoardCreator/main.js

```diff
@@ -0,0 +1,175 @@
+			 * 0 is up-left, 1 is up-center, and so on.
+			 */
+			console.log("Creating entry for cellIndex: " + cellIndex + " at r" + currR + " c" + currC);
+			adjacencyMatrix[cellIndex][0] = (currR === 0 || currC === 0) ? -1 : this.cellIndices[currR-1][currC-1];
+			adjacencyMatrix[cellIndex][1] = (currR === 0) ? -1 : this.cellIndices[currR-1][currC];
+			adjacencyMatrix[cellIndex][2] = (currR === 0 || currC === this.cols - 1) ? -1 : this.cellIndices[currR-1][currC+1];
+			adjacencyMatrix[cellIndex][3] = (currC === 0) ? -1 : this.cellIndices[currR][currC-1];
+			adjacencyMatrix[cellIndex][4] = cellIndex;
+			adjacencyMatrix[cellIndex][5] = (currC === this.cols - 1) ? -1 : this.cellIndices[currR][currC+1];
+			adjacencyMatrix[cellIndex][6] = (currR === this.rows - 1 || currC === 0) ? -1 : this.cellIndices[currR+1][currC-1];
+			adjacencyMatrix[cellIndex][7] = (currR === this.rows - 1) ? -1 : this.cellIndices[currR+1][currC];
+			adjacencyMatrix[cellIndex][8] = (currR === this.rows - 1 || currC === this.cols - 1) ? -1 : this.cellIndices[currR+1][currC+1];
+		}
+		return adjacencyMatrix;
+	}
+	
+	createAdjacencyMatrixString()
+	{
+		const adjMatrix = this.createAdjacencyMatrix();
+		let adjMatStr = "[";
+		for (let i = 0; i < adjMatrix.length; i++)
+		{
+			adjMatStr += "[" + adjMatrix[i].toString() + "],";
+		}
+		adjMatStr = adjMatStr.slice(0, -1) + "]";
+		return adjMatStr;
+	}
+}
+
+/* Initialize main variables */
+let canvas = document.getElementById("canvas");
+let ctx = canvas.getContext("2d");
+ctx.font = "18px Arial";
+let cellSize = 50;
+let board = new MockBoard(8, 8);
+updateBoard();
+updateOutput();
+
+function updateBoardSize()
+{
+	const rows = parseInt(document.getElementById("rows").value);
+	const cols = parseInt(document.getElementById("cols").value);
+	board.setRows(rows);
+	board.setCols(cols);
+	updateBoard();
+}
+
+function updateCanvas()
+{
+	ctx.clearRect(0, 0, canvas.width, canvas.height);
+	cellSize = parseInt(document.getElementById("gridwidth").value);
+	const rows = parseInt(document.getElementById("rows").value);
+	const cols = parseInt(document.getElementById("cols").value);
+	canvas.width = cols * cellSize;
+	canvas.height = rows * cellSize;
+}
+
+function drawGridLines(color)
+{
+	let ctx = canvas.getContext("2d");
+	ctx.strokeStyle = "#BBBBBB";
+	for(let i = 0; i < canvas.width; i+=cellSize)
+	{
+		ctx.beginPath();
+		ctx.moveTo(i, 0);
+		ctx.lineTo(i, canvas.height);
+		ctx.stroke();
+	}
+	for(let i = 0; i < canvas.height; i+=cellSize)
+	{
+		ctx.beginPath();
+		ctx.moveTo(0, i);
+		ctx.lineTo(canvas.width, i);
+		ctx.stroke();
+	}
+}
+
+function updateBoard()
+{
+	updateCanvas();
+	board.updateCellIndexList();
+	for (let r = 0; r < board.rows; r++)
+	{
+		for (let c = 0; c < board.cols; c++)
+		{
+			if (board.cells[r][c])
+			{
+				ctx.fillStyle = "#00FF00";
+			}
+			else
+			{
+				ctx.fillStyle = "#FF0000";
+			}
+			ctx.fillRect(c*cellSize, r*cellSize, cellSize - 3, cellSize - 3);
```

Comment:
`totallynothardcoded.jpg`

Reply:
Hush. It looks fine when you load the page.